src/node_crypto.cc

```diff
@@ -2990,9 +2991,15 @@ void SSLWrap<Base>::CertCbDone(const FunctionCallbackInfo<Value>& args) {
     goto fire_cb;
 
   if (cons->HasInstance(ctx)) {
-    SecureContext* sc;
-    ASSIGN_OR_RETURN_UNWRAP(&sc, ctx.As<Object>());
-    w->sni_context_.Reset(env->isolate(), ctx);
+    SecureContext* sc = Unwrap<SecureContext>(ctx.As<Object>());
+    CHECK_NOT_NULL(sc);
+    // XXX: There is a method w->SetSNIContext(sc), and you might think that
+    // it makes sense to call that here and make setting w->sni_context_ part
+    // of it. In fact, that passes the test suite, although SetSNIContext()
+    // performs a lot more operations.
+    // If anybody is familiar enough with the TLS code to know whether it makes
+    // sense, please do so or document why it doesn't.
+    w->sni_context_ = BaseObjectPtr<SecureContext>(sc);
```

Comment:
@nodejs/crypto … if anybody happens to know more about this, feel free to enlighten me :)

Reply:
`SetSNIContext()` is what you call when you're ready to start using the the context. That's not the case here, it's just stored for later use.

`SetSNIContext()` has only one caller (in tls_wrap.cc) so perhaps it's a good idea to inline it at the call site and remove the method to avoid future confusion.

Reply:
> `SetSNIContext()` has only one caller (in tls_wrap.cc) so perhaps it's a good idea to inline it at the call site and remove the method to avoid future confusion.

@bnoordhuis Done in 03251c56531b0c1dc4387bb51008c29e14e53b51, I think this is about what you had in mind?

Reply:
Yep, that's what I meant.